### ISSUE TITLE
how to change the range of small, medium, large in the result of val?

### ISSUE THREAD

**Issue by lilichu** [facebookresearch/Detectron#572]
helllo!
I want change the range of small, medium, large
```
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=  8 ] = 0.422
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=  8 ] = 0.746
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=  8 ] = 0.414
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=  8 ] = 0.251
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=  8 ] = 0.445
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=  8 ] = 0.604
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.257
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  5 ] = 0.489
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  8 ] = 0.493
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=  8 ] = 0.337
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=  8 ] = 0.521
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=  8 ] = 0.643
```
I want change small, medium, large from: 
```
        [0**2, 32**2],     # small
        [32**2, 96**2],    # medium
        [96**2, 1e5**2],   # large
```
to:
```
        # [0 ** 2, 10 ** 2],  # small
        # [10 ** 2, 40 ** 2],  # medium
        # [40 ** 2, 1e5 ** 2],  # large
```
how to change them? thanks!

**Comment by ir413** [comment]
Hi @lilichu, object area ranges are defined in the COCO API eval params (see [here](https://github.com/cocodataset/cocoapi/blob/master/PythonAPI/pycocotools/cocoeval.py#L510)). You can change the params field of the COCO eval object (see [here](https://github.com/cocodataset/cocoapi/blob/master/PythonAPI/pycocotools/cocoeval.py#L76)). In detectron, COCO eval objects are constructed in the [`json_dataset_evaluator`](https://github.com/facebookresearch/Detectron/blob/master/detectron/datasets/json_dataset_evaluator.py) (e.g. [here](https://github.com/facebookresearch/Detectron/blob/master/detectron/datasets/json_dataset_evaluator.py#L193) for detection evaluation).

**Comment by lilichu** [comment]
thanks！
If I change it in the cocoapi files, the result doesn't change:
`/home/user/lichu/COCOAPI/PythonAPI/pycocotools/cocoeval.py`
When I debug, I find the `cocoeval.py` is in:
`/home/user/.PyCharm2018.1/system/remote_sources/-964779609/pycocotools-2.0-py2.7-linux-x86_64.egg!/pycocotools/cocoeval.py`
so I need change the params in this file path?

**Comment by ir413** [comment]
Depending on how you installed the COCO API you may need to re-install it after making changes to it. However, I think it is easier not to make changes to the COCO API directly but instead set/change the `coco_eval.params` field from the detectron code (see my previous comment for pointers). I realize that this is not very clean but I don't think that the COCO API provides a generic way for doing this.

**Comment by ash1995** [comment]
I had a couple of questions regarding the area ranges:

How were these limits chosen? I mean how was it decided that objects with area in between 0 and 32^2 will be categorised as small, objects with area in between 32^2 and 96^2 will be categorised as medium etc. This would most likely have been done based on some canonical image size. If so, what was that image size?

Why is the upper limit of large object areas set to a really high value of 1e5^2? I believe it is to capture any objects larger than 96^2 but wouldn't that hurt the AP score? For example, in one of my projects I was using the default ranges for evaluating my object detector. Once I reduced the upper limit from 1e5^2 to 256^2, I got a boost in my AP score. This was most likely due to me not having any objects with area above 256^2. I am not sure about this finding, so kindly correct me if I'm mistaken.
